File: CS-499 Software Engineering and Design/Original/investment.cpp
/*
 * Investment.cpp
 *
 *  Date: 06/03/2023
 *  Author: Ryan Summers
 */

#include <iostream>
#include <iomanip>

using namespace std;

// Set variables
float initialAmount;
float monthlyDeposit;
float annualInterest;
float years;
float months;
float totalAmount;
float interestAmount;
float totalYear;

int main()
{
	// Menu
	cout << "************************************\n";
	cout << "************ DATA INPUT ************\n";
	cout << "Initial Investment Amount: \n";
	cout << "Monthly Deposit: \n";
	cout << "Annual Interest: \n";
	cout << "Number of years: \n";
	system("PAUSE");
	// Inputs
	cout << "************************************\n";
	cout << "************ DATA INPUT ************\n";
	cout << "Initial Investment Amount: $";
	cin >> initialAmount;
	cout << "Monthly Deposit: $";
	cin >> monthlyDeposit;
	cout << "Annual Interest: %";
	cin >> annualInterest;
	cout << "Number of years: ";
	cin >> years;
	months = years * 12;
	system("PAUSE");

	// Initial amount
	totalAmount = initialAmount;
	cout << "\n   Balance and Interest Without Additional Monthly Deposits\n";
	cout << "================================================================\n";
	cout << "Year\t\tYear End Balance\tYear End Earned Interest\n";
	cout << "----------------------------------------------------------------\n";

	// Figure total interest
	for (int i = 0; i < years; i++)
	{
		// Yearly interest
		interestAmount = totalAmount * (annualInterest / 100);
		// Total year total
		totalAmount = totalAmount + interestAmount;
		// set fixed and 2 decimal places
		cout << (i + 1) << "\t\t$" << fixed << setprecision(2) << totalAmount << "\t\t\t$" << interestAmount << "\n";
	}

	// Total
	totalAmount = initialAmount;

	// Deposits
	cout << "\n\n   Balance and Interest With Additional Monthly Deposits\n";
	cout << "================================================================\n";
	cout << "Year\t\tYear End Balance\tYear End Earned Interest\n";
	cout << "----------------------------------------------------------------\n";

	// Interst and deposits 
	for (int i = 0; i < years; i++)
	{
		// yearly interest zero percent at the begining of the year
		totalYear = 0;
		for (int j = 0; j < 12; j++)
		{
			// monthly interest
			interestAmount = (totalAmount + monthlyDeposit) * ((annualInterest / 100) / 12);
			// month end interest
			totalYear = totalYear + interestAmount;
			// month end total 
			totalAmount = totalAmount + monthlyDeposit + interestAmount;
		}
		cout << (i + 1) << "\t\t$" << fixed << setprecision(2) << totalAmount << "\t\t\t$" << totalYear << "\n";
	}
	return 0;
}

File: CS-499 Software Engineering and Design/Enhancement/investment.py
# Program description
print("This is an interest calculating program")
print("You will need to input the initial amount,")
print("the amount you will deposit each month, the")
print("annual interest rate, and the number of years.")
print("===============================================")
print("The program will show you the balance and interest,")
print("without additional deposits, and the amount with")
print("additional deposits.")
print("")

# Inputs
print("====================================")
print("         Input data below")
print("====================================")
initial_amount = float(input("Initial Investment: $"))
monthly_deposit = float(input("Monthly Deposit: $"))
annual_interest = float(input("Annual Interest: %"))
years = float(input("Number of years: "))
months = years * 12

# Initial amount
total_amount = initial_amount
print("\n   Balance and Interest Without Additional Monthly Deposits")
print("================================================================")
print("Year\t\tYear End Balance\tYear End Earned Interest")
print("----------------------------------------------------------------")

# Figure total interest
for i in range(int(years)):
    # Yearly interest
    interest_amount = total_amount * (annual_interest / 100)
    # Total year total
    total_amount += interest_amount
    # set fixed and 2 decimal places
    print(f"{i + 1}\t\t\t\t${total_amount:.2f}\t\t\t\t${interest_amount:.2f}")

# Total
total_amount = initial_amount

# Deposits
print("\n\n   Balance and Interest With Additional Monthly Deposits")
print("================================================================")
print("Year\t\tYear End Balance\tYear End Earned Interest")
print("----------------------------------------------------------------")

# Interest and deposits
for i in range(int(years)):
    # yearly interest zero percent at the beginning of the year
    total_year = 0
    for j in range(12):
        # monthly interest
        interest_amount = (total_amount + monthly_deposit) * ((annual_interest / 100) / 12)
        # month end interest
        total_year += interest_amount
        # month end total
        total_amount += monthly_deposit + interest_amount
    print(f"{i + 1}\t\t\t\t${total_amount:.2f}\t\t\t\t${total_year:.2f}")
